src/doc/switcher.inc

```diff
@@ -0,0 +1 @@
+<script defer src='https://doc.rust-lang.org/version-switcher.js'></script>
```

Comment:
Per https://github.com/rust-lang/rust/pull/101855/files#r973608362, I think this file needs to be slightly templated as wel, ideally a relative path so we support dev-doc well -- basically, I'd like it if we just land *this* PR we get the right behavior on nightly immediately, and in all future Rust releases. We still want something like a trivial include mechanism so that it's easy to insert it into past HTML, but this PR isn't there yet.

Reply:
The include on rustdoc side is unfortunately limited. For example, we can't use a relative path for this JS file as it wouldn't work on `file:///`. To work for all previous versions, as we discussed, we either need to regenerate the documentation or to make an insertion when serving the files. We could also add this JS directly into the `main.js` script in all versions and that would work too. Might be ever simpler.

Reply:
If appending to main.js works consistently (and identifying that file is easy, which is probably true), then I think that seems like a good option to me, since it side-steps some of the difficulty here.

Reply:
There is one down-side putting this script into `main.js` though: starting `1.63` (with [this commit](https://github.com/rust-lang/rust/commit/88787a33a7325e5444e0151c6a75558b0c514234)), the `main.js` script import is `defer`, meaning it doesn't block loading the page, meaning the top navbar will appear after the page is loaded, making a weird update. Not a blocker but worth mentioning I think.

So do you want me to simply insert it at the start of the `main.js` file or do you want to keep it into another file?